modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,486 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.appserver.integration.lazy.loading;
+
+import org.apache.axis2.AxisFault;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.appserver.integration.common.clients.WebAppAdminClient;
+import org.wso2.appserver.integration.common.utils.ASIntegrationTest;
+import org.wso2.appserver.integration.lazy.loading.util.LazyLoadingTestException;
+import org.wso2.appserver.integration.lazy.loading.util.TenantStatus;
+import org.wso2.appserver.integration.lazy.loading.util.WebAppStatus;
+import org.wso2.carbon.application.mgt.stub.ApplicationAdminExceptionException;
+import org.wso2.carbon.authenticator.stub.LoginAuthenticationExceptionException;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.test.utils.common.TestConfigurationProvider;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import org.wso2.carbon.automation.test.utils.http.client.HttpURLConnectionClient;
+import org.wso2.carbon.integration.common.admin.client.ApplicationAdminClient;
+import org.wso2.carbon.integration.common.utils.LoginLogoutClient;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.carbon.utils.ServerConstants;
+import org.xml.sax.SAXException;
+
+import javax.xml.stream.XMLStreamException;
+import javax.xml.xpath.XPathExpressionException;
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.rmi.RemoteException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Base class to hold the common variables and utility methods of Lazy Loading (Ghost deployment) Test cases.
+ * All Lazy Loading (Ghost deployment)  Test  Classes must inherit from this.
+ * Replacing the carbon.xml with Ghost deployment configuration and deployment of the
+ * tenant-info-service web application is implemented in side init() method.
+ */
+public abstract class LazyLoadingBaseTest extends ASIntegrationTest {
+
+    protected static final String TENANT_DOMAIN_1_KEY = "tenant1";
+    protected static final String TENANT_DOMAIN_2_KEY = "tenant2";
+    protected static final String SUPER_TENANT_DOMAIN_KEY = "superTenant";
+    protected static final String CARBON_HOME = System.getProperty(ServerConstants.CARBON_HOME);
```

Comment:
Maybe it's a good idea to only depend on Util functionalities exposed by CarbonUtil and so on, to retrieve info like this.